generic_hw_control_loop:
  loop_hz: 200
  cycle_time_error_threshold: 0.01

# Settings for ros_control hardware interface (used in generic_hw_interface.cpp)
hardware_interface:
  joints:
    - robot_arm_mount_to_actuator
    - joint2_to_joint1
    - joint3_to_joint2
    - joint4_to_joint3
    - joint5_to_joint4
    - joint6_to_joint5
    - joint6output_to_joint6

# Publish all joint states 
# creates the /joint_states topic necessary in ROS
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 200


arm_motion_group_controller:
  type: position_controllers/JointTrajectoryController
  joints:
    - robot_arm_mount_to_actuator
    - joint2_to_joint1
    - joint3_to_joint2
    - joint4_to_joint3
    - joint5_to_joint4
    - joint6_to_joint5
    - joint6output_to_joint6
  # gains:
  #   joint2_to_joint1: {p: 100, d: 1, i: 1, i_clamp: 1}
  #   joint3_to_joint2: {p: 100, d: 1, i: 1, i_clamp: 1}
  #   joint4_to_joint3: {p: 100, d: 1, i: 1, i_clamp: 1}
  #   joint5_to_joint4: {p: 100, d: 1, i: 1, i_clamp: 1}
  #   joint6_to_joint5: {p: 100, d: 1, i: 1, i_clamp: 1}
  #   joint6output_to_joint6: {p: 100, d: 1, i: 1, i_clamp: 1}
